src/riot-rs-embassy/src/lib.rs

```diff
@@ -4,34 +4,23 @@
 #![feature(impl_trait_in_assoc_type)]
 #![feature(used_with_arg)]
 #![feature(lint_reasons)]
-#![feature(trait_alias)]
+#![feature(doc_auto_cfg)]
 
 pub mod define_peripherals;
 pub mod gpio;
 
-#[cfg(feature = "external-interrupts")]
-mod extint_registry;
-
-#[cfg(context = "cortex-m")]
-pub mod executor_swi;
-
 cfg_if::cfg_if! {
```

Comment:
Can we add `package = "riot-rs-arch"` in Cargo.toml (e.g., [here](https://github.com/future-proof-iot/RIOT-rs/pull/392/files#diff-edd9eb0f4bbee9a8df8c5d0250ce7091e80fe2f20160190bf476a0641e69ed3cR48)) and replace this whole cfg_if with `pub use riot_rs_arch as arch;`?

Reply:
> We can simplify the cfg_if by merging the first four branches using package as you said (and I think that would indeed be better),